src/api/routers/participants/participantsCreation.ts

```diff
@@ -16,68 +16,61 @@ import { getKcAdminClient } from '../../keycloakAdminClient';
 import { addSite, getSiteList, setSiteClientTypes } from '../../services/adminServiceClient';
 import {
   mapClientTypesToAdminEnums,
-  SiteCreationRequest,
+  SiteCreationRequest
 } from '../../services/adminServiceHelpers';
 import {
   constructAuditTrailObject,
-  performAsyncOperationWithAuditTrail,
+  performAsyncOperationWithAuditTrail
 } from '../../services/auditTrailService';
 import {
-  assignApiParticipantMemberRole,
-  createNewUser,
-  sendInviteEmailToNewUser,
+  doesUserExistInKeycloak
 } from '../../services/kcUsersService';
 import {
   getParticipantTypesByIds,
   ParticipantRequest,
-  sendNewParticipantEmail,
+  sendNewParticipantEmail
 } from '../../services/participantsService';
-import { findUserByEmail } from '../../services/usersService';
+import {
+  createAndInviteKeycloakUser,
+  findUserByEmail,
+  sendInviteEmailToExistingUser
+} from '../../services/usersService';
 import {
   ParticipantCreationAndApprovalPartial,
-  ParticipantCreationRequest,
+  ParticipantCreationRequest
 } from './participantClasses';
 
 export async function validateParticipantCreationRequest(
   participantRequest: z.infer<typeof ParticipantCreationRequest>
 ) {
-  let errorMessage = null;
   const existingParticipant = await Participant.query().findOne(
     'name',
     participantRequest.participantName
   );
   if (existingParticipant) {
-    errorMessage = 'Duplicate participant name';
-  }
-  const existingUser = await findUserByEmail(participantRequest.email);
-  if (existingUser) {
-    errorMessage = 'Duplicate requesting user';
-  }
-  const kcAdminClient = await getKcAdminClient();
-  const existingKcUser = await kcAdminClient.users.find({ email: participantRequest.email });
-  if (existingKcUser.length > 0) {
-    errorMessage = 'Requesting user already exists in Keycloak';
+    return 'Duplicate participant name';
   }
-
   if (!participantRequest.siteId) {
     // check for duplicate site in admin
     const { siteName } = participantRequest;
     // this is inefficient but we'd need a new endpoint in admin to search by name
     const sites = await getSiteList();
     if (sites.filter((site) => site.name === siteName).length > 0) {
-      errorMessage = 'Requested site name already exists';
+      return 'Requested site name already exists';
     }
   }
-  return errorMessage;
+  return null;
 }
 
-const createUserAndAssociatedParticipant = async (
+const createParticipantWithUser = async (
   parsedUser: z.infer<typeof UserCreationPartial>,
   participantData: z.infer<typeof ParticipantCreationAndApprovalPartial>
-) => {
-  await User.transaction(async (trx) => {
-    // create user
-    const newPortalUser = await User.query(trx).insertAndFetch(parsedUser);
+): Promise<Participant | undefined> => {
+  const participant = await User.transaction(async (trx) => {
+    let user = await findUserByEmail(parsedUser.email);
+    if (!user) {
+      user = await User.query(trx).insertAndFetch(parsedUser);
```

Comment:
Only insert/update the user if they don't exist already. Otherwise leave them as-is (don't update their name/jobFunction)